### PR TITLE
internal link lesson - remove obsolete name attribute

### PR DESCRIPTION
**Pull request by marigerr**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue #15672

#### Description

According to [W3C docs](https://www.w3.org/TR/html5/obsolete.html#obsolete) and [MDN](https://developer.mozilla.org/en/docs/Web/HTML/Element/a#Obsolete)  using the `name` attribute on anchor tags is obsolete and `id` should be used instead.

PR changes the [lesson on Internal Links](http://beta.freecodecamp.com/en/challenges/basic-html-and-html5/link-to-internal-sections-of-a-page-with-anchor-elements) to demonstrate creating an internal link to jump to any element by using `a` tag's `href` to reference the target element's` id`.  Removed references to `name` attribute.  Also in general tried to simplify/shorten the lesson as it was longer than other lessons in this section.

Thanks in advance for review and feedback!

Edit - Forgot to mention that this was brought to attention due to an [issue ](https://github.com/freeCodeCamp/testable-projects-fcc/issues/143) opened on _testable-projects-fcc_ repo.   In the test suite, there is a test that checks for an `id `of a internal link target element instead of a `name`.   It seemed better to update the lesson than change the test after learning that `a` tag `name ` attribute is obsolete.

### CODE REVIEW

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 1]
```diff
@@ -759,18 +759,15 @@
...
       "description": [
-        "In addition to creating external links, anchor elements can also be used to create internal links, which are links that jump to different sections within a webpage.",
-        "The format is similar to an external link except instead of a URL, you'll use the <code>#</code> and a word to describe the section you want to jump to.",
-        "Here's an example:",
-        "<code>&lt;a href=\"#contact\"&gt;Go to contact section&lt;/a&gt;</code>",
-        "Next, you'll want to create a corresponding anchor link in the HTML where you want the internal link to take users. Instead of using the <code>href</code> attribute, you'll use the <code>name</code> attribute. Here's an example:",
-        "<blockquote>&lt;h1&gt;Contact&lt;/h1&gt;<br>&lt;a name=\"contact\"&gt;&lt;/a&gt;</blockquote>",
-        "Now when users click the \"Go to contact section\" link, they'll be taken to the section of the webpage with the anchor that has the <code>name</code> attribute \"contact\".",
+        "Anchor or <code>a</code> elements can also be used to create an internal link to jump to different section within a webpage.",
```
Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: "different sections"

instead of:

"different section"

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 2]
```diff
@@ -759,18 +759,15 @@
...
+        "To create an internal link, the <code>a</code> tag's <code>href</code> attribute contains the hash symbol <code>#</code> and the <code>id</code> of the target element to jump to.  An <code>id</code> is an attribute that uniquely describes an element.",
```
Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I think I would rephrase this line to:
"To create an internal link, you assign a link's `href` value to a hash symbol (`#`) plus the id for an element that you want to internally link to, usually farther down the page. You would then need to add the same `id` attribute to the element you are linking to."

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 3]
```diff
@@ -783,12 +780,16 @@
         "  <p>Kitty ipsum dolor sit amet, shed everywhere shed everywhere stretching attack your ankles chase the red dot, hairball run catnip eat the grass sniff. Purr jump eat the grass rip the couch scratched sunbathe, shed everywhere rip the couch sleep in the sink fluffy fur catnip scratched. Kitty ipsum dolor sit amet, shed everywhere shed everywhere stretching attack your ankles chase the red dot, hairball run catnip eat the grass sniff.</p>",
         "  <p>Purr jump eat the grass rip the couch scratched sunbathe, shed everywhere rip the couch sleep in the sink fluffy fur catnip scratched. Kitty ipsum dolor sit amet, shed everywhere shed everywhere stretching attack your ankles chase the red dot, hairball run catnip eat the grass sniff. Purr jump eat the grass rip the couch scratched sunbathe, shed everywhere rip the couch sleep in the sink fluffy fur catnip scratched.</p>",
```
Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Let's add two more paragraphs of Kitty Ipsum text here so the effect of clicking the link and jumping to another section is clear.

Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Perhaps, they are generated here? http://www.catipsum.com/ 

If not, just copy/paste the two current paragraphs again.

**Review by dhcodes — CHANGES_REQUESTED** [review]
See all my comments.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 2 above)
Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Thanks! I knew that sentence was awkward, your version sounds much better.   What do you think about my second sentence about id.     I felt the need to include because id's hadn't been brought up in prior lessons yet.

> An id is an attribute that uniquely describes an element.

Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I like it and vote to keep it. But you do have two spaces after the `.` so I would make that one space.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 4]
```diff
@@ -759,18 +759,15 @@
...
-        "Don't forget to remove the <code>target=\"_blank\"</code> attribute since this causes the linked document to open in a new window tab.",
-        "Then add an anchor link with a <code>name</code> attribute set to \"bottom\" after the last paragraph.",
-        "<strong>Note</strong><br>Additional kitty ipsum text has been added to the paragraphs to better show the effect of clicking the internal anchor link to jump to the bottom of the page. Also, this change will only apply to this challenge. The next challenge goes back to showing your external link."
+        "Change your external link to an internal link by changing the <code>href</code> attribute to \"#footer\" and the text from \"cat photos\" to \"Jump to Bottom\".",
+        "Remove the <code>target=\"_blank\"</code> attribute from the anchor tag since this causes the linked document to open in a new window tab.",
+        "Then add an <code>footer</code> element at the bottom of the page after the <code>main</code> element.  Set its <code>id</code> to \"footer\"."
```
Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: One space after periods

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 1 above)
Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I wonder if it might work to add the singular "a" 

> jump to a different section within a webpage.

For some reason to me it sounded confusing for it to be plural there, like its sounds like one link could take you to several different places vs one link/one destination.

Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I went ahead and changed this sentence to be all plural!

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 3 above)
Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: sounds good! I included 2 new paragraphs in my PR update

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 5]
```diff
@@ -759,18 +759,15 @@
...
-        "Next, you'll want to create a corresponding anchor link in the HTML where you want the internal link to take users. Instead of using the <code>href</code> attribute, you'll use the <code>name</code> attribute. Here's an example:",
-        "<blockquote>&lt;h1&gt;Contact&lt;/h1&gt;<br>&lt;a name=\"contact\"&gt;&lt;/a&gt;</blockquote>",
-        "Now when users click the \"Go to contact section\" link, they'll be taken to the section of the webpage with the anchor that has the <code>name</code> attribute \"contact\".",
+        "Anchor or <code>a</code> elements can also be used to create internal links to jump to different sections within a webpage.",
+        "To create an internal link, you assign a link's <code>href</code> attribute to a hash symbol <code>#</code> plus the <code>id</code> attribute for an element that you want to internally link to, usually further down the page. You would then need to add the same <code>id</code> attribute to the element you are linking to. An <code>id</code> is an attribute that uniquely describes an element.",
+        "Below is an example of an internal anchor link and it's target element:",
```
Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: "it's" should be "its"

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 6]
```diff
@@ -759,18 +759,15 @@
...
-        "Don't forget to remove the <code>target=\"_blank\"</code> attribute since this causes the linked document to open in a new window tab.",
-        "Then add an anchor link with a <code>name</code> attribute set to \"bottom\" after the last paragraph.",
-        "<strong>Note</strong><br>Additional kitty ipsum text has been added to the paragraphs to better show the effect of clicking the internal anchor link to jump to the bottom of the page. Also, this change will only apply to this challenge. The next challenge goes back to showing your external link."
+        "Change your external link to an internal link by changing the <code>href</code> attribute to \"#footer\" and the text from \"cat photos\" to \"Jump to Bottom\".",
+        "Remove the <code>target=\"_blank\"</code> attribute from the anchor tag since this causes the linked document to open in a new window tab.",
+        "Then add an <code>footer</code> element at the bottom of the page after the <code>main</code> element. Set its <code>id</code> to \"footer\"."
```
Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: "Then add an" should be "then add a"

Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I would make the last sentence "Give it an `id` attribute of "footer". 

Picky, I know!

**Review by dhcodes — CHANGES_REQUESTED** [review]

**Comment by dhcodes** [comment]
A few more changes. @no-stack-dub-sack. You're a curriculum guy. What do you think about these changes? I need a second pair of eyes.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 6 above)
Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I really appreciate your review!  I updated the PR with the latest 3 changes

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 7]
```diff
@@ -759,18 +759,15 @@
...
-        "Don't forget to remove the <code>target=\"_blank\"</code> attribute since this causes the linked document to open in a new window tab.",
-        "Then add an anchor link with a <code>name</code> attribute set to \"bottom\" after the last paragraph.",
-        "<strong>Note</strong><br>Additional kitty ipsum text has been added to the paragraphs to better show the effect of clicking the internal anchor link to jump to the bottom of the page. Also, this change will only apply to this challenge. The next challenge goes back to showing your external link."
+        "Change your external link to an internal link by changing the <code>href</code> attribute to \"#footer\" and the text from \"cat photos\" to \"Jump to Bottom\".",
+        "Remove the <code>target=\"_blank\"</code> attribute from the anchor tag since this causes the linked document to open in a new window tab.",
+        "Then add a <code>footer</code> element at the bottom of the page after the <code>main</code> element. Give it an <code>id</code> attribute of \"footer\"."
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I'm concerned that "after the `main` element" may be interpreted as "after the `<main>` opening tag" or "as the first child of the `main` element". Perhaps changing this to "after the closing `</main>` tag" could help avoid that confusion.

Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: This is the first time in the curriculum a camper is asked to use the `footer` element. It's only mentioned once before. I'm concerned that this might make this challenge more complicated than it needs to be. Maybe we should make them jump to the last paragraph? Or we add a paragraph at the bottom that says "bottom", which they then add the id to?

Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]:
```diff
- `id` attribute of "footer"
+ `id` attribute with a value of "footer"
```

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 8]
```diff
@@ -782,13 +779,19 @@
...
+        "  <p>Meowwww loved it, hated it, loved it, hated it yet spill litter box, scratch at owner, destroy all furniture, especially couch or lay on arms while you're using the keyboard. Missing until dinner time toy mouse squeak roll over. With tail in the air lounge in doorway. Man running from cops stops to pet cats, goes to jail.</p>",
+        "  <p>Intently stare at the same spot poop in the plant pot but kitten is playing with dead mouse. Get video posted to internet for chasing red dot leave fur on owners clothes meow to be let out and mesmerizing birds leave fur on owners clothes or favor packaging over toy so purr for no reason. Meow to be let out play time intently sniff hand run outside as soon as door open yet destroy couch.</p>",                
         "  ",
-        "</main>"
+        "</main>",
+        "  "
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: For consistency, there shouldn't be an empty line at the end right here. I like having empty lines at the end, but I like consistency more 😄

**Review by systimotic — CHANGES_REQUESTED** [review]
I really like this change! This is definitely something that needs to happen. I left a few comments with suggestions for minor improvements to minimize possible confusion.
Nice work @marigerr! 👍

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 3 above)
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: @dhcodes Is there a way to mark this as completed?

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 7 above)
Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I agree with this.

Comment (dhcodes) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I thought about this too, but wasn't sure if it'd be a big deal since there's a main element.

**Review by dhcodes — APPROVED** [review]

**Comment by dhcodes** [comment]
Thanks @systimotic

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 7 above)
Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Thanks @systimotic for review!  
If we keep the footer element, could change the instructions to add brackets around \<footer\> element.  This would make it more intuitive what a footer element is.
But changing to jump to last paragraph is a good idea also.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 8 above)
Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: I think I included the extra line so it would be possibly be more obvious that that is where the new footer element should go.   But can remove it if looks messy. :)

**Comment by marigerr** [comment]
Ok I did a couple more updates hopefully this is almost good to go!   I wanted to double check on an remaining _requested change_ -  about removing an extra line at end of seed code.   I was thinking that it might be good to leave it in so that the user has a convenient and obvious place to add there code.

**Comment by dhcodes** [comment]
@marigerr I agree that we should leave the spacing. I know this already feels like the never-ending PR but could you fix something unrelated to this challenge? On [this line](https://github.com/freeCodeCamp/freeCodeCamp/pull/15682/files#diff-70c67b41bf89de779a2b07e9d67d96f9R778) can you remove the extra space before the `\"`. There's an extra space after the word "back". 

@systimotic I think this is good to go. You?

**Comment by marigerr** [comment]
@dhcodes  I realize it's essentially re-writing the lesson, so want to make sure it's correct!  I really appreciate your and @systimotic help.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 8 above)
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Good call 👍

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 7 above)
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: This comment collapsed, but it doesn't look like this has been changed. Do you know what happened @marigerr?

Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: The `main` element is introduced quite extensively, but I feel like campers won't have learned about enough HTML elements at this point that they'll understand what to do when a `<footer>` element is thrown at them. For that reason, I'm in favour of adding the jump to an existing element. I think we could also consider adding the footer for them.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 9]
```diff
@@ -759,36 +759,39 @@
...
-        "assert($('a').eq(1).attr('name') == \"bottom\", 'message: The second <code>a</code> tag should have a <code>name</code> attribute set to bottom.');"
+        "assert($('a').length == 1, 'message: There should be only one anchor tag on your page.');",
+        "assert($('footer').length == 1, 'message: There should be only one <code>footer</code> tag on your page.');",
+        "assert($('a').eq(0).attr('href') == \"#footer\", 'message: The <code>a</code> tag should have an <code>href</code> attribute set to \"#footer\".');",
+        "assert(typeof $('a').eq(0).attr('target') == typeof undefined || $('a').eq(0).attr('target') == true, 'message: The <code>a</code> tag should not have an <code>target</code> attribute');",        
+        "assert($('a').eq(0).text().match(/Jump to Bottom/gi), 'message: The <code>a</code> text should be \"Jump to Bottom\".');",        
+        "assert($('footer').eq(0).attr('id') == \"footer\", 'message: The <code>footer</code> tag should have a <code>id</code> attribute set to \"footer\".');"
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]:
```diff
- a <code>id</code>
+ an <code>id</code>
```

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 10]
```diff
@@ -759,36 +759,39 @@
...
       "description": [
-        "In addition to creating external links, anchor elements can also be used to create internal links, which are links that jump to different sections within a webpage.",
-        "The format is similar to an external link except instead of a URL, you'll use the <code>#</code> and a word to describe the section you want to jump to.",
-        "Here's an example:",
-        "<code>&lt;a href=\"#contact\"&gt;Go to contact section&lt;/a&gt;</code>",
-        "Next, you'll want to create a corresponding anchor link in the HTML where you want the internal link to take users. Instead of using the <code>href</code> attribute, you'll use the <code>name</code> attribute. Here's an example:",
-        "<blockquote>&lt;h1&gt;Contact&lt;/h1&gt;<br>&lt;a name=\"contact\"&gt;&lt;/a&gt;</blockquote>",
-        "Now when users click the \"Go to contact section\" link, they'll be taken to the section of the webpage with the anchor that has the <code>name</code> attribute \"contact\".",
+        "Anchor or <code>a</code> elements can also be used to create internal links to jump to different sections within a webpage.",
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: "Anchor or `a`" suggests that they're two separate things. The previous challenge mentions what they are and that they are the same, so I think going with only "Anchor" is fine here. Using the original sentence would also be OK.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 11]
```diff
@@ -759,36 +759,39 @@
...
+        "To create an internal link, you assign a link's <code>href</code> attribute to a hash symbol <code>#</code> plus the <code>id</code> attribute for an element that you want to internally link to, usually further down the page. You would then need to add the same <code>id</code> attribute to the element you are linking to. An <code>id</code> is an attribute that uniquely describes an element.",
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: 1. "hash symbol `#` plus the `id` attribute"
This is a similar to the other place I mentioned this in my first look at this. Officially the attribute is the entire thing: `id="potato"`. You're referring specifically to the `potato` part, therefore, I think you should say "the value of the `id` attribute"

2. "You would then need to add"
There a change here from "you do this" to "you would do this". I'd like to keep the "you do this" tone, which could be achieved by removing "would" here.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 12]
```diff
@@ -759,36 +759,39 @@
...
+        "Below is an example of an internal anchor link and its target element:",
+        "<blockquote>&lt;a href=\"#contactsHeader\"&gt;Contacts&lt;/a&gt;<br>...<br>&lt;h2 id=\"contactsHeader\"&gt;Contacts&lt;/h2&gt;</blockquote>",
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: This example is great! In the CSS section, all of our class names and ID's are hyphen-delimited. For consistency, I think it would be a good idea to do that here too.

**Review by systimotic — CHANGES_REQUESTED** [review]
I'm sorry for commenting on things now that were the same when I first looked at this, it annoys me when people do that. For some reason, I decided only to look at 20% of this the first time. The other 80% is already really good though 👍 Only a few minor suggestions.

Test look great, but I haven't tested them to see if they do what they're supposed to.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 7 above)
Comment (marigerr) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Hmm, strange this is showing changed for me!   Can you double check?  @systimotic

**Comment by marigerr** [comment]
@systimotic Ok great, I believe I got all the updates made.

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json (hunk 7 above)
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Yep, shows changed now! Sorry about that, must've been a caching issue.

**Review by systimotic — APPROVED** [review]
Code LGTM 👍 👍 

Really nice work @marigerr

I haven't had time to test this challenge, and I won't have time to do so for the next two weeks, so if @dhcodes can confirm it works the way it's supposed to I'd say this is good to be merged 👍

**Comment by dhcodes** [comment]
**LGTM**. I like adding the footer from the get-go and having it be the copyright. Good idea @marigerr. Thanks for making all the changes. 👍 💯 🌮 🦅

**Comment by marigerr** [comment]
@dhcodes @systimotic
Ah fantastic - yay team!  
Btw, there is a Portuguese translation for this lesson that needs to be updated to reflect the new content.
I think I will open up an issue for that, and possibly CC/mention the original translator to see if they are possibly interested in updating it..
